support/cas-server-support-pm-webflow/src/main/java/org/apereo/cas/pm/web/flow/PasswordManagementSingleSignOnParticipationStrategy.java

```diff
@@ -0,0 +1,38 @@
+package org.apereo.cas.pm.web.flow;
+
+import org.apereo.cas.ticket.registry.TicketRegistry;
+import org.apereo.cas.web.flow.SingleSignOnParticipationStrategy;
+
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * This is {@link PasswordManagementSingleSignOnParticipationStrategy}.
+ *
+ * @author Julien Huon
+ * @since 6.3.0
+ */
+@RequiredArgsConstructor
+public class PasswordManagementSingleSignOnParticipationStrategy implements SingleSignOnParticipationStrategy {
+    private final TicketRegistry ticketRegistry;
```

Comment:
You should be using `CentralAuthenticationService`, and then catch exceptions when you get the ticket. Also, please add logging statements under debug/trace to the method here to explain the behavior.

Reply:
Will do it. :)

I'm a bit curious. Is it related to performance to catch an exception rather than have a null response ?

Reply:
No, it has to do with transactional semantics. Fetching a ticket id `TicketRegistry` without a running transaction can cause issues for registry implementations that require transactions like those backed by JDBC and a relational database. `CentralAuthenticationService` helps with that.

Reply:
Understood. Will remember it for future work.

Thanks

Reply:
Done